tests/test_tree.py

```diff
@@ -138,12 +143,14 @@ def _get_best_split(self, X, y):
                 t = (v + values[i + 1]) * 0.5
                 groups = (vals <= t).astype(int)
                 if groups.sum() >= min_leaf and (X.shape[0] - groups.sum()) >= min_leaf:
-                    s, _ = compare_survival(y, groups)
-                    if s > best_stat:
+                    s, p = compare_survival(y, groups)
+                    abs_z = abs(norm.ppf(p/2))
+                    if s > best_stat and abs_z >= min_logrank:
```

Comment:
The `LogrankCriterion` class uses the absolute value of the log-rank test statistic as splitting criterion, whereas here, the condition is based on the p-value.